src/dotnet/commands/dotnet-install/dotnet-install-tool/InstallToolCommand.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.DotNet.Cli;
+using Microsoft.DotNet.Cli.CommandLine;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Configurer;
+using Microsoft.DotNet.ShellShimMaker;
+using Microsoft.DotNet.ToolPackageObtainer;
+using Microsoft.Extensions.EnvironmentAbstractions;
+
+namespace Microsoft.DotNet.Tools.Install.Tool
+{
+    public class InstallToolCommand : CommandBase
+    {
+        private static string _packageId;
+        private static string _packageVersion;
+        private static string _configFilePath;
+        private static string _framework;
+
+        public InstallToolCommand(
+            AppliedOption appliedCommand,
+            ParseResult parseResult)
+            : base(parseResult)
+        {
+            if (appliedCommand == null)
+            {
+                throw new ArgumentNullException(nameof(appliedCommand));
+            }
+
+            _packageId = appliedCommand.Arguments.Single();
+            _packageVersion = appliedCommand.ValueOrDefault<string>("version");
+            _configFilePath = appliedCommand.ValueOrDefault<string>("configfile");
+            _framework = appliedCommand.ValueOrDefault<string>("framework");
+        }
+
+        public override int Execute()
+        {
+            FilePath? configFile = null;
+
+            if (_configFilePath != null)
+            {
+                configFile = new FilePath(_configFilePath);
+            }
+
+            var executablePackagePath = new DirectoryPath(new CliFolderPathCalculator().ExecutablePackagesPath);
+
+            var toolConfigurationAndExecutableDirectory = ObtainPackage(
+                _packageId,
+                _packageVersion,
+                configFile,
+                _framework,
+                executablePackagePath);
+
+            DirectoryPath executable = toolConfigurationAndExecutableDirectory
+                .ExecutableDirectory
+                .WithSubDirectories(
+                    toolConfigurationAndExecutableDirectory
+                        .Configuration
+                        .ToolAssemblyEntryPoint);
+
+            var shellShimMaker = new ShellShimMaker.ShellShimMaker(executablePackagePath.Value);
```

Comment:
This caught my eye before. I would prefer if we changed either the namespace or the class name. This pattern is ugly.

Reply:
fixing...

Reply:
done